t/mojolicious/003_users.t

```diff
@@ -72,14 +75,37 @@ my $added_user_to_course = {
 	role    => 'student'
 };
 $t->post_ok('/webwork3/api/courses/4/users' => json => $added_user_to_course)->status_is(200)
-	->content_type_is('application/json;charset=UTF-8')->json_is('/role' => 'student');
+	->content_type_is('application/json;charset=UTF-8')->json_is('/role' => 'STUDENT');
+
+my $lisa = firstval { $_->{username} eq 'lisa' } @all_users;
+
+# Check if the user is a global user
+$t->get_ok('/webwork3/api/courses/1/users/lisa/exists')->status_is(200)
+	->content_type_is('application/json;charset=UTF-8')->json_is('/first_name' => $lisa->{first_name});
+
+# Check if a non-existent user is a global user
+$t->get_ok('/webwork3/api/courses/1/users/non_existent_user/exists')->status_is(200)
+	->content_type_is('application/json;charset=UTF-8');
+
+is_deeply($t->tx->res->json, {}, 'checkUserExists: check that a non-existent user returns {}');
+
+# Testing that booleans returned from the server are JSON booleans.
+# the first user is the admin
+
+my $admin_user = firstval { $_->{username} eq 'admin' } @all_users;
```

Comment:
Something for the future:
I don't really like the usage of the `List::MoreUtils` module for only the `firstval` method as you have done all over in the tests and this is the only way this module is ever used.  That is something that can be done quite easily with native perl.  Just change this to
```suggestion
my $admin_user = ({ $_->{username} eq 'admin' } @all_users)[0];
```
That code is equivalent.  It returns the first element of the array, and is undefined if the array is empty.

If this is changed, that is one less module that we need as a dependency.

Reply:
Thanks for this.  I didn't know about this.

Reply:
Actually, don't we need a grep here, that is:
```
my $admin_user = (grep { $_->{username} eq 'admin' } @all_users)[0];
```